c++/src/kj/memory.h

```diff
@@ -22,9 +22,29 @@
 #pragma once
 
 #include "common.h"
-#ifdef KJ_DEBUG
-#include <atomic>  // std::atomic for KJ_ASSERT_PTR_COUNTERS
-#endif
+
+// KJ_DEBUG_MEMORY == 1 enables variety of checks designed to catch memory usage errors.
+// KJ_DEBUG_MEMORY == 0 disables all such checks.
+// When not set it is initialized to 1 in KJ_DEBUG mode, 0 otherwise.
```

Comment:
We probably don't want KJ_DEBUG to change the ABI as people commonly build (and install) release libs, then build debug binaries against them. I think KJ_DEBUG_MEMORY does affect the ABI, so we'll have to have it off by default even under KJ_DEBUG I think.

Reply:
Redid not to enable `KJ_DEBUG_MEMORY` by default. Also updated the CI to exercise debug_memory.